langchain4j/src/main/java/dev/langchain4j/service/DefaultAiServices.java

```diff
@@ -216,8 +124,7 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
                             return handleChatMemoryAccess(method, args);
                         }
 
-                        // TODO do it once, when creating AI Service?
-                        validateParameters(context.aiServiceClass, method);
+                        context.validateParameters(method);
```

Comment:
```suggestion
                        // TODO do it once, when creating AI Service?
                        context.validateParameters(method);
```
I would still keep the comment as it is better to validate it as soon as possible to fail fast if something is invalid, instead of waiting for the first time AI Service is used

Reply:
I originally tried to perform the validation at service build time, but this made a few test to fail, and more important would have introduced a backward incompatibility that, has discussed, we want to avoid it for now. I will restore the todo comment, even though the problem is now mitigated by the cache that I introduced.

Reply:
Done.